### ISSUE TITLE
feature request: api expansion to allow for tools like Rekit Studio

### ISSUE THREAD

**Issue by shaun-sweet** [Microsoft/vscode#43530]
So I am wondering what the VScode team thinks of a tool such as https://github.com/supnate/rekit

A demo can be found http://demo.rekit.org/

I absolutely love it.  I would LOVE to be able to have something like this... but inside of VScode, not a web based IDE.  However, it seems the author has looked into doing it specifically in VScode but found the API too limiting to allow it.  [See this issue](https://github.com/supnate/rekit/issues/74)

Basically at its core, its virtual folders and virtual tabs (and somewhat nested tabs)

Thoughts on getting support for the ability to do things like this?

**Comment by mjbvz** [comment]
This would make a good extension. Please file individual issues for any limitations you are running into with the current VS Code API
